podman/domain/containers.py

```diff
@@ -170,10 +170,32 @@ def exec_run(
             NotImplementedError: method not implemented.
             APIError: when service reports error
         """
-        if user is None:
-            user = "root"
-
-        raise NotImplementedError()
+        user = user or "root"
+        if isinstance(environment, dict):
+            environment = [f"{k}={v}" for k, v in environment.items()]
+        data = {
+            "AttachStderr": stderr,
+            "AttachStdin": stdin,
+            "AttachStdout": stdout,
+            "Cmd": cmd if isinstance(cmd, list) else [cmd],
+            # "DetachKeys": detach,  # This is something else
+            "Env": environment,
+            "Privileged": privileged,
+            "Tty": tty,
+            "User": user,
+            "WorkingDir": workdir
+        }
+        # create the exec instance
+        response = self.client.post(f"/containers/{self.name}/exec", data=json.dumps(data))
+        response.raise_for_status()
+        exec_id = response.json()['Id']
+        # start the exec instance
+        response = self.client.post(f"/exec/{exec_id}/start", params={"Detach": detach, "Tty": tty})
```

Comment:
@JacobCallahan Thanks for your work on this. https://docs.podman.io/en/latest/_static/api.html#operation/ExecStartLibpod shows instead of `params` you want to pass that dictionary as JSON using the `data` keyword.